yugabyte/src/yugabyte/ysql/client.clj

```diff
@@ -32,6 +32,49 @@
    :connectTimeout (/ default-timeout 1000)
    :socketTimeout  (/ default-timeout 1000)})
 
+(defn query
+  "Like jdbc query, but includes a default timeout in ms.
```

Comment:
Can we just set timeout at the connection level and avoid specifying it in every request?

Reply:
This ends up in `java.sql.Statement.setQueryTimeout(int)` for `PreparedStatement`.
Here's the interface method with its comment from JDK:
```java
    /**
     * Sets the number of seconds the driver will wait for a
     * <code>Statement</code> object to execute to the given number of seconds.
     *By default there is no limit on the amount of time allowed for a running
     * statement to complete. If the limit is exceeded, an
     * <code>SQLTimeoutException</code> is thrown.
     * A JDBC driver must apply this limit to the <code>execute</code>,
     * <code>executeQuery</code> and <code>executeUpdate</code> methods.
     * <p>
     * <strong>Note:</strong> JDBC driver implementations may also apply this
     * limit to {@code ResultSet} methods
     * (consult your driver vendor documentation for details).
     * <p>
     * <strong>Note:</strong> In the case of {@code Statement} batching, it is
     * implementation defined as to whether the time-out is applied to
     * individual SQL commands added via the {@code addBatch} method or to
     * the entire batch of SQL commands invoked by the {@code executeBatch}
     * method (consult your driver vendor documentation for details).
     *
     * @param seconds the new query timeout limit in seconds; zero means
     *        there is no limit
     * @exception SQLException if a database access error occurs,
     * this method is called on a closed <code>Statement</code>
     *            or the condition {@code seconds >= 0} is not satisfied
     * @see #getQueryTimeout
     */
    void setQueryTimeout(int seconds) throws SQLException;
```
I see no reference to the connect/socket timeout set on the connection level, nor could I find (at a glance, haven't dug deep) a relation between them in PG driver code. As such, I propose to keep it this way.